tests/chainer_tests/training_tests/extensions_tests/test_parameter_statistics.py

```diff
@@ -3,68 +3,166 @@
+        'report_params': False,
+        'report_grads': True,
+        'prefix': 'test',
+        'expect': 2
+    },
+    {
+        'links': [chainer.links.Linear(3, 2)],
+        'statistics': {'min': lambda x: cuda.get_array_module(x).min(x)},
+        'report_params': True,
+        'report_grads': False,
+        'prefix': None,
+        'expect': 2
+    },
+    {
+        'links': [chainer.links.Linear(3, 2)],
+        'statistics': {'min': lambda x: cuda.get_array_module(x).min(x)},
+        'report_params': False,
+        'report_grads': False,
+        'prefix': 'test',
+        'expect': 0
+    },
+    {
+        'links': [chainer.links.Linear(3, 2)],
+        'statistics': extensions.ParameterStatistics.default_statistics,
+        'report_params': True,
+        'report_grads': True,
+        'prefix': None,
+        'expect': 36
+    },
+    {
+        'links': [chainer.links.Linear(3, 2)],
+        'statistics': extensions.ParameterStatistics.default_statistics,
+        'report_params': True,
+        'report_grads': False,
+        'prefix': 'test',
+        'expect': 24
+    },
+    {
+        'links': [chainer.links.Linear(3, 2)],
+        'statistics': extensions.ParameterStatistics.default_statistics,
+        'report_params': False,
+        'report_grads': True,
+        'prefix': None,
+        'expect': 12
+    },
+    {
+        'links': [chainer.links.Linear(3, 2)],
+        'statistics': extensions.ParameterStatistics.default_statistics,
+        'report_params': False,
+        'report_grads': False,
+        'prefix': 'test',
+        'expect': 0
+    }
 )
 class TestParameterStatistics(TestParameterStatisticsBase, unittest.TestCase):
 
     def test_report(self):
-        extension = extensions.ParameterStatistics(self.links,
-                                                   statistics=self.statistics)
-        self.trainer.extend(extension)
+        self.trainer.extend(self.create_extension())
         self.trainer.run()
-
         self.assertEqual(len(self.trainer.observation), self.expect)
 
     def test_report_late_register(self):
-        extension = extensions.ParameterStatistics(self.links, statistics={})
+        extension = self.create_extension(skip_statistics=True)
         for name, function in six.iteritems(self.statistics):
             extension.register_statistics(name, function)
         self.trainer.extend(extension)
         self.trainer.run()
-
         self.assertEqual(len(self.trainer.observation), self.expect)
 
     def test_report_key_pattern(self):
-        extension = extensions.ParameterStatistics(self.links)
-        self.trainer.extend(extension)
+        self.trainer.extend(self.create_extension())
         self.trainer.run()
 
         pattern = r'^(.+/){2,}(data|grad)/.+[^/]$'
         for name in six.iterkeys(self.trainer.observation):
+            if self.report_params and self.report_grads:
+                assert any(attr in name for attr in ['data', 'grad'])
```

Comment:
Doesn't this check covered with regex check?

Reply:
Removed the redundant check.